tests/test_Sequence.cc

```diff
@@ -3911,3 +3911,77 @@ TEST_CASE("Sequence: Check setter/getter function for step setup script", "[Sequ
         REQUIRE(ctx.step_setup_script == script);
     }
 }
+
+TEST_CASE("Sequence: sequence timeout", "[Sequence]")
+{
+    Step step{Step::type_action};
+    step.set_script("sleep(0.2)");
+
+    Sequence seq{"test_sequence"};
+    seq.push_back(std::move(step));
+
+    seq.set_timeout(100ms);
+    REQUIRE(seq.get_timeout() == 100ms);
+
+    Context ctx;
+    REQUIRE(seq.get_time_of_last_execution().time_since_epoch().count() == 0L);
+    auto maybe_error = seq.execute(ctx, nullptr);
+    REQUIRE(seq.get_time_of_last_execution().time_since_epoch().count() != 0L);
+
+    REQUIRE(maybe_error.has_value() == true);
+    REQUIRE_THAT(maybe_error->what(), Contains("Timeout: Sequence"));
+}
+
+TEST_CASE("Sequence: sequence timeout is lower than step timeout", "[Sequence]")
```

Comment:
Maybe we should drop this test case. If I am not mistaken, it does not test anything more than the previous one, except the exact number of milliseconds that have been waited - but unfortunately that can not really be tested because execution times are not deterministic. I'd rather save this 0.5 seconds.